### PR TITLE
TRACING-3279: Add documentation for Batch processor for OTEL

### PR DESCRIPTION
**Pull request by iblancasa**
Version(s):
4.11, 4.12, 4.13, 4.14

Issue:
https://issues.redhat.com/browse/TRACING-3279

Link to docs preview:

QE review:
- [x] QE has approved this change.

Additional information:

@IshwarKanse 
@max-cx

### CODE REVIEW

File: modules/distr-tracing-otel-config-collector.adoc [hunk 1]
```diff
@@ -239,6 +239,43 @@ The Zipkin receiver ingests data in the Zipkin v1 and v2 formats.
...
+
+[id="batch-processor_{context}"]
+==== Batch processor
+
+The Resource Detection processor is designed to identify host resource details in alignment with OpenTelemetry's resource semantic standards. Using this detected information, it can add or replace the resource values in telemetry data.
+
+* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]
+* Supported signals: traces, metrics
+
+.OpenTelemetry Collector using the Resource Detection processor
```
Comment (andreasgerstmayr) [inline modules/distr-tracing-otel-config-collector.adoc:251]:
```suggestion
.OpenTelemetry Collector using the Batch processor
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 2]
```diff
@@ -239,6 +239,43 @@ The Zipkin receiver ingests data in the Zipkin v1 and v2 formats.
 [id="processors_{context}"]
 === Processors
 
+
+[id="batch-processor_{context}"]
+==== Batch processor
+
+The Resource Detection processor is designed to identify host resource details in alignment with OpenTelemetry's resource semantic standards. Using this detected information, it can add or replace the resource values in telemetry data.
```
Comment (andreasgerstmayr) [inline modules/distr-tracing-otel-config-collector.adoc:246]: Looks like a copy&paste error?

Comment (max-cx) [reply modules/distr-tracing-otel-config-collector.adoc:246]: Thank you for spotting this, Andreas.

Comment (iblancasa) [reply modules/distr-tracing-otel-config-collector.adoc:246]: Sorry. It seems when I moved the PR from one branch to another I messed up

File: modules/distr-tracing-otel-config-collector.adoc [hunk 3]
```diff
@@ -239,6 +239,43 @@ The Zipkin receiver ingests data in the Zipkin v1 and v2 formats.
...
+
+[id="batch-processor_{context}"]
+==== Batch processor
+
+The batch processor batches the data to reduce the number of outgoing connections needed to transfer the telemetry information.
+
+* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]
+* Supported signals: traces, metrics
+
+.OpenTelemetry Collector using the Batch processor
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:251]:
```suggestion
Example of the .OpenTelemetry Collector custom resource when using the Batch processor
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 4]
```diff
@@ -239,6 +239,43 @@ The Zipkin receiver ingests data in the Zipkin v1 and v2 formats.
...
+      pipelines:
+        traces:
+          processors: [batch]
+        metrics:
+          processors: [batch]
+----
+
+.Parameters used by the Batch processor
+[cols="3",options="header"]
+|============================================================================================================================================================================================================
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:269]:
```suggestion
|===
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 5]
```diff
@@ -239,6 +239,43 @@ The Zipkin receiver ingests data in the Zipkin v1 and v2 formats.
...
+| send_batch_size            | Send the batch of telemetry data after the desired number of spans or metrics                                                                                          | 8192 
+| send_batch_max_size        | The maximum allowable size of the batch. Must be equal or greater than send_batch_size.                                                                                | 0    
+| metadata_keys              | When activated, this processor will generate a batcher instance for each unique set of values found in the client.Metadata.                                            | []   
+| metadata_cardinality_limit | When the metadata_keys are populated, this configuration restricts the number of distinct metadata key value combinations processed throughout the process's duration. | 1000 
+|============================================================================================================================================================================================================
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:276]:
```suggestion
|===
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 6]
```diff
@@ -239,6 +239,43 @@ The Zipkin receiver ingests data in the Zipkin v1 and v2 formats.
...
+|Parameter |Description |Default
+| timeout                    | Send the batch after a specific time duration, irrespective of its size.                                                                                               | 200ms
+| send_batch_size            | Send the batch of telemetry data after the desired number of spans or metrics                                                                                          | 8192 
+| send_batch_max_size        | The maximum allowable size of the batch. Must be equal or greater than send_batch_size.                                                                                | 0    
+| metadata_keys              | When activated, this processor will generate a batcher instance for each unique set of values found in the client.Metadata.                                            | []   
+| metadata_cardinality_limit | When the metadata_keys are populated, this configuration restricts the number of distinct metadata key value combinations processed throughout the process's duration. | 1000
```
[the remark this answers is not in the document]
Comment (iblancasa) [reply modules/distr-tracing-otel-config-collector.adoc:275]: > BTW, I didn't understand the word batcher

Even the upstream documentation used that term:
https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/batchprocessor/README.md

It is like: you have a batch but, if you uses that parameter, you will have multiple "batch queues". I'm totally open to change it but I was not able to find a better wording. Any idea?

**Comment by iblancasa** [comment]
@max-cx is there anything else to be done?

File: modules/distr-tracing-otel-config-collector.adoc [hunk 7]
```diff
@@ -239,6 +239,57 @@ The Zipkin receiver ingests data in the Zipkin v1 and v2 formats.
...
+| Sends the batch after a specific time duration, irrespective of its size.
+| 200ms
+
+| `send_batch_size`
+| Sends the batch of telemetry data after the specified number of spans or metrics.
+| 8192
+
+| `send_batch_max_size`
+| The maximum allowable size of the batch. Must be equal or greater than `send_batch_size`.
+| 0
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:282]: I'm curious--what are the units for this parameter?

File: modules/distr-tracing-otel-config-collector.adoc [hunk 8]
```diff
@@ -239,6 +239,57 @@ The Zipkin receiver ingests data in the Zipkin v1 and v2 formats.
...
+        send_batch_max_size: 10000
+    service:
+      pipelines:
+        traces:
+          processors: [batch]
+        metrics:
+          processors: [batch]
+----
+
+.Parameters used by the Batch processor
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:267]:
```suggestion
.Parameters used by the batch processor
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 9]
```diff
@@ -239,6 +239,57 @@ The Zipkin receiver ingests data in the Zipkin v1 and v2 formats.
...
+
+[id="batch-processor_{context}"]
+==== Batch processor
+
+The batch processor batches the data to reduce the number of outgoing connections needed to transfer the telemetry information.
+
+* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]
+* Supported signals: traces, metrics
+
+.Example of the OpenTelemetry Collector custom resource when using the Batch processor
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:251]:
```suggestion
.Example of the OpenTelemetry Collector custom resource when using the batch processor
```

File: modules/distr-tracing-otel-config-collector.adoc (hunk 8 above)
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:267]: Looks like it's a lowercase "batch processor":
https://opentelemetry.io/search/?q=batch%20processor
https://github.com/open-telemetry/opentelemetry-collector/tree/main/processor/batchprocessor
```suggestion
.Parameters used by the batch processor
```

File: modules/distr-tracing-otel-config-collector.adoc (hunk 9 above)
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:251]: Looks like it's a lowercase "batch processor":
https://opentelemetry.io/search/?q=batch%20processor
https://github.com/open-telemetry/opentelemetry-collector/tree/main/processor/batchprocessor
```suggestion
.Example of the OpenTelemetry Collector custom resource when using the batch processor
```

File: modules/distr-tracing-otel-config-collector.adoc (hunk 8 above)
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:267]: Let's put it in lowercase for now (I'll review all these names later):
```suggestion
.Parameters used by the batch processor
```

File: modules/distr-tracing-otel-config-collector.adoc (hunk 7 above)
Comment (iblancasa) [reply modules/distr-tracing-otel-config-collector.adoc:282]: Number of spans, metric data points, or log records

File: modules/distr-tracing-otel-config-collector.adoc [hunk 10]
```diff
@@ -245,6 +245,57 @@ The Zipkin receiver ingests data in the Zipkin v1 and v2 formats.
...
+| `timeout`
+| Sends the batch after a specific time duration, irrespective of its size.
+| 200ms
+
+| `send_batch_size`
+| Sends the batch of telemetry data after the specified number of spans or metrics.
+| 8192
+
+| `send_batch_max_size`
+| The maximum allowable size of the batch in terms of the number of spans, metric data points, or log records. Must be equal or greater than `send_batch_size`.
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:287]:
```suggestion
| The maximum allowable size of the batch in terms of the number of spans, metric data points, or log records. Must be equal to or greater than `send_batch_size`.
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 11]
```diff
@@ -245,6 +245,57 @@ The Zipkin receiver ingests data in the Zipkin v1 and v2 formats.
...
+| 0
+
+| `metadata_keys`
+| When activated, a batcher instance is created for each unique set of values found in the `client.Metadata`.
+| []
+
+| `metadata_cardinality_limit`
+| When the `metadata_keys` are populated, this configuration restricts the number of distinct metadata key-value combinations processed throughout the duration of the process.
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:295]:
```suggestion
| When the `metadata_keys` are populated, this configuration restricts the number of processed distinct metadata key-value combinations during the process.
```

**Comment by iblancasa** [comment]
@max-cx is there anything else to be done here?

**Review by IshwarKanse — APPROVED** [review]

**Review by max-cx — APPROVED** [review]
LGTM.

**Comment by stevsmit** [comment]
PR looks good to me. The commits need squashed, though. Please squash and re-submit.

**Comment by stevsmit** [comment]
lgtm

**Pull request merged**